compiler/parser/parser.cpp

```diff
@@ -328,7 +328,28 @@ static void parseCommandLineFiles() {
   }
 
   while ((inputFileName = nthFilename(fileNum++))) {
-    if (isChplSource(inputFileName)) {
+    if (isChplSource(inputFileName))
+    {
+      /*
+      Ensure that all the files parsed don't exceed the
+      (NAME_MAX - 16) i.e. 239 bytes Limits
+      */
+      const size_t maxFileName = NAME_MAX - 16;
+      if (strlen(inputFileName) > maxFileName)
+      {
+        char reducedFileName[maxFileName];
+        // sprintf fileName to allowed length buffer
+        strncpy(reducedFileName, inputFileName, maxFileName);
+        // error message format with fileName and allowed length
+        const char *errorFormat = "%s, filename is longer than maximum allowed length of %d";
+        // allow max length in error with error
+        const size_t errorLength = maxFileName + strlen(errorFormat) - 4 + 3;
+        // buffer to store errorMessage
+        char errorMessage[errorLength];
+        snprintf(errorMessage, errorLength, errorFormat, reducedFileName, maxFileName);
+        // throw error
+        USR_FATAL(errorMessage);
```

Comment:
I think you should be able to just do `USR_FATAL(errorFormat, inputFilename, maxFileName);` here?  (`USR_FATAL()` should be a printf-like function.  This will also save you lots of pain around computing `reducedFileName`, `errorLength`, etc.).